### PR TITLE
Display general error message when have exception SN-712

### PR DESCRIPTION
**Pull request by tanhieu-nguyen**
https://saritasa.atlassian.net/browse/SN-712

### CODE REVIEW

File: src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs [hunk 1]
```diff
@@ -358,13 +358,22 @@ private static Expression GetConvertedExpressionWhenPropertyIsNotString(MemberEx
...
+            // We use Attach instead of Add because
+            // EF will try to create new entity and create all navigation (even when they are exist in database).
+            // Attach resolves this problem by explicitly attaching navigation to EF change tracker.
+            dbContext.Attach(entity);
+            dbContext.Add(entity);
+            await dbContext.SaveChangesAsync(cancellationToken);
+        }
+        catch (Exception)
+        {
+            dbContext.ChangeTracker.Clear();
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs:372]: Please add comment why we need this

File: src/Saritasa.NetForge.Mvvm/ViewModels/EditEntity/EditEntityViewModel.cs [hunk 2]
```diff
@@ -149,7 +152,17 @@ public async Task UpdateEntityAsync()
...
+            IsUpdated = true;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "{handler}: {message}", nameof(EditEntityViewModel), ex.Message);
+
+            GeneralError = ex.InnerException is not null ? ex.InnerException.Message : ex.Message;
+        }
+
     }
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.Mvvm/ViewModels/EditEntity/EditEntityViewModel.cs:167]: Please fix warning on this line

File: src/Saritasa.NetForge.Mvvm/ViewModels/ValidationEntityViewModel.cs [hunk 3]
```diff
@@ -6,7 +6,12 @@
 public abstract class ValidationEntityViewModel : BaseViewModel
 {
     /// <summary>
-    /// List of <see cref="FieldErrorModel"/> instances in the view model.
+    /// General error message.
+    /// </summary>
+    public string GeneralError { get; set; } = "";
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.Mvvm/ViewModels/ValidationEntityViewModel.cs:11]: `string.Empty`?

**Review by AlexanderMakarov124 — CHANGES_REQUESTED** [review]
1. Please add screenshots to see how it looks
2. Do we need to handle exceptions when we remove entity?

**Comment by tanhieu-nguyen** [comment]
Create entity or edit entity:
- image

Delete entity:
- image

**Comment by toanphamsaritasa** [comment]
Can we have similar style when displaying exceptions for create, update and delete? Create or edit is using MudAlert and delete is using Snackbar is just strange. Can we just use MudAlert or Snackbar for all of these operations?

**Comment by AlexanderMakarov124** [comment]
> Can we have similar style when displaying exceptions for create, update and delete? Create or edit is using MudAlert and delete is using Snackbar is just strange. Can we just use MudAlert or Snackbar for all of these operations?

To choose between `MudAlert` and `Snackbar` I prefer to use `MudAlert` because `Snackbar` is hiding after some time.

**Comment by tanhieu-nguyen** [comment]
Updated:
Create entity or edit entity:

image

**Review by vasiliy-chefonov — APPROVED** [review]

**Review by toanphamsaritasa — APPROVED** [review]
Please resolve the conflicts

**Pull request merged**